Makefile

```diff
@@ -38,42 +35,59 @@ TIMEOUT:=15
 #
 # See https://bts.frama-c.com/view.php?id=2206
 
+SLEVEL_FUNCTIONS:=_extract_complex_tag:100, \
+	_parse_arc:100, \
+	parse_UTCTime:100, \
+	find_dn_by_oid:100, \
+	find_curve_by_oid:100, \
+	find_alg_by_oid:200, \
+	find_ext_by_oid:200, \
+	parse_AccessDescription:400, \
+	parse_x509_Extension:400, \
+	parse_x509_subjectPublicKeyInfo:300, \
+	parse_x509_Extensions:400, \
+	bufs_differ:200, \
+	parse_x509_tbsCertificate:400, \
+	parse_x509_AlgorithmIdentifier:200
+
+FRAMACOPTS:=-machdep x86_64 \
+	-warn-left-shift-negative \
+	-warn-right-shift-negative \
+	-warn-signed-downcast \
+	-warn-signed-overflow \
+	-warn-unsigned-downcast \
+	-warn-unsigned-overflow \
+	-rte \
+	-eva \
+	-wp-dynamic \
+	-eva-slevel 1 \
+	-slevel-function="$(SLEVEL_FUNCTIONS)" \
+	-eva-warn-undefined-pointer-comparison none \
+	-then \
+	-wp \
+	-wp-dynamic \
+	-wp-par $(JOBS) \
+	-wp-steps 100000 -wp-depth 100000 -pp-annot \
+	-wp-split -wp-literals \
+	-wp-timeout $(TIMEOUT)
+
 frama-c:
-	frama-c-gui src/x509-parser.c -machdep x86_64 \
-	            -warn-left-shift-negative \
-	            -warn-right-shift-negative \
-	            -warn-signed-downcast \
-	            -warn-signed-overflow \
-	            -warn-unsigned-downcast \
-	            -warn-unsigned-overflow \
-		    -rte \
-		    -eva \
-		    -wp-dynamic \
-		    -eva-slevel 1 \
-		    -slevel-function="_extract_complex_tag:100, \
-				      _parse_arc:100, \
-				      parse_UTCTime:100, \
-				      find_dn_by_oid:100, \
-				      find_curve_by_oid:100, \
-				      find_alg_by_oid:200, \
-				      find_ext_by_oid:200, \
-				      parse_AccessDescription:400, \
-				      parse_x509_Extension:400, \
-				      parse_x509_subjectPublicKeyInfo:300, \
-				      parse_x509_Extensions:400, \
-				      bufs_differ:200, \
-				      parse_x509_tbsCertificate:400, \
-				      parse_x509_AlgorithmIdentifier:200" \
-		    -eva-warn-undefined-pointer-comparison none \
-		    -then \
-		    -wp \
-		    -wp-dynamic \
-		    -wp-par $(JOBS) \
-		    -wp-steps 100000 -wp-depth 100000 -pp-annot \
-		    -wp-split -wp-literals \
-		    -wp-timeout $(TIMEOUT) -save $(SESSION)
+	$(FRAMAC) src/x509-parser.c $(FRAMACOPTS) -save $(SESSION)
 
 frama-c-gui:
-	frama-c-gui -load $(SESSION)
+	$(FRAMACGUI) src/x509-parser.c $(FRAMACOPTS) -save $(SESSION)
+
+test:
+	$(FRAMAC) src/x509-parser.c $(FRAMACOPTS) > verif.log
+	@fgrep -e 'Proved goals' verif.log > .proved_goals
+	git diff --exit-code -- .proved_goals; cat verif.log && rm -f verif.log
```

Comment:
Compare first, then output the verification log. Travis-CI has a 10000 lines limit for an output.